moveit_core/collision_detection_fcl/src/collision_common.cpp

```diff
@@ -531,14 +535,14 @@ bool distanceCallback(fcl::CollisionObjectd* o1, fcl::CollisionObjectd* o2, void
   {
     return false;
   }
-  double d = fcl::distance(o1, o2, fcl::DistanceRequestd(cdata->req->enable_nearest_points), fcl_result);
+  double distance = fcl::distance(o1, o2, fcl::DistanceRequestd(cdata->req->enable_nearest_points), fcl_result);
 
   // Check if either object is already in the map. If not add it or if present
   // check to see if the new distance is closer. If closer remove the existing
   // one and add the new distance information.
-  if (d < dist_threshold)
+  if (distance < dist_threshold)
   {
-    DistanceResultsData dist_result;
+    thread_local DistanceResultsData dist_result;
```

Comment:
This can actually benefit from thread_local storage, because `Eigen::Vector3d` allocates memory on the heap, which can be reused.

Reply:
> because Eigen::Vector3d allocates memory on the heap

[This statement is wrong](https://eigen.tuxfamily.org/dox/group__TutorialMatrixClass.html) and confused me quite a bit.
Fixed-size Eigen structures use local storage where possible. They can produce some overhead though when compiled with less than `-O3`?
The problem are the `std::string` link names in this case and it still makes sense to keep this variable thread_local.

Reply:
Thanks for pointing out the correct Eigen doc link. I was quickly searching the web yesterday and found a generic statement somewhere else stating the opposite. AND I somehow missed the string array.